src/components/PostsList.tsx

```
@@ -1,86 +1,70 @@
-import React from 'react';
+import { FC } from 'react';
+import { Post } from '../types/Post';
+import classNames from 'classnames';
 
-export const PostsList: React.FC = () => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+type Props = {
+  posts: Post[];
+  activePost: Post | null;
+  getActivePost: (post: Post | null) => void;
+};
 
-    <table className="table is-fullwidth is-striped is-hoverable is-narrow">
-      <thead>
-        <tr className="has-background-link-light">
-          <th>#</th>
-          <th>Title</th>
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <th> </th>
-        </tr>
-      </thead>
+export const PostsList: FC<Props> = ({
+  posts,
+  activePost,
+  getActivePost = () => {},
```

You provide a default for `getActivePost` in the destructuring: `getActivePost = () => {}` even though the prop is typed as required in `Props`. This default can hide bugs when the parent forgets to pass the handler. Either remove the default so missing prop errors surface, or update the `Props` type to mark `getActivePost` optional if that is intentional. Keeping these consistent improves type-safety and catches integration mistakes earlier.